Transformation from hardcoded elements to a fully configurable system.

Altrix-One opened the pull request:
<img width="1470" height="799" alt="image" src="https://github.com/user-attachments/assets/1a488815-28eb-4419-8462-c4545d4d8bc0" />

<img width="1470" height="801" alt="image" src="https://github.com/user-attachments/assets/732044c8-cf34-4666-9ecb-d3604d0f2e38" />


<img width="1470" height="604" alt="image" src="https://github.com/user-attachments/assets/4a613f3b-5e8c-48bd-bfbf-143c7c8474c1" />


<img width="1470" height="525" alt="image" src="https://github.com/user-attachments/assets/1c1e28e1-4766-41c5-95e4-21e8a8b9f243" />


**Major Accomplishments:**

- Dynamic Store Branding - Configurable store name, logos, and favicon
- Comprehensive Color System - 11 different color controls including primary, accent, UI elements, and footer colors
- Flexible Footer Management - Database-driven footer sections with nested links
- Contact & Social Media Integration - Configurable contact info and social platform links
- Email Template System - Automated order confirmations, stock alerts, and cancellations

**Technical Implementation:**

- 3 New Doctypes: Footer Section Config, Footer Link, Ecommerce Category
- 25+ New Fields in Lifestyle Settings across 4 organized tabs
- CSS Custom Properties system for real-time color changes
- Template Updates across 8+ files replacing hardcoded classes
- Migration Scripts with default data population

**Key Benefits:**

- No Code Changes Required - Full customization through ERPNext UI
- Real-time Updates - Changes apply immediately
- Brand Consistency - Centralized control over all visual elements
- Maintainable Architecture - Clean separation of configuration and presentation

Altrix-One committed:
Issues Identified and Fixed:
Missing Required Custom Fields

Added custom_displayname and custom_item_group_display_name to all Item Group creations
Missing Root Item Group

Created get_root_item_group() function to dynamically find or create the root
Root Item Group Also Missing Custom Fields

Updated root Item Group creation (lines 84-85) to include both required fields

Altrix-One committed:
Fixed Landing Page IndexError ✅
Issue: Products list page crashed with IndexError: list index out of range when no brands exist.

Fixed in list.py:

Changed if not brands[0]: to if not brands: (line 27)
Added filter to remove None/empty brand values: brands = [b.title() for b in brands if b] (line 29)
This prevents the error when:

No products have brands assigned
Brand field is empty/None
Fresh installation with no products yet
The landing page should now load correctly even without any products or brands in the system.

Altrix-One committed:
Demo Data and Fixes

Altrix-One committed:
Dynamic Item Groups

Altrix-One committed:
Branding

Altrix-One committed:
update script

Altrix-One committed:
Logos

Altrix-One committed:
Update Settings

Altrix-One committed:
Update Settings with colors

Altrix-One committed:
Cleanup

Altrix-One committed:
Update Footer Sections

Altrix-One committed:
Fix Footer Linking

NagariaHussain commented:
Hi 👋 

This looks awesome! Can you please add some screenshots or video of the frontend after applying this customisations?

NagariaHussain commented on ls_shop/migrate.py:
```diff
@@ -27,24 +34,245 @@ def create_payment_modes():
+		"Engine Parts": "Engine Parts",
+		"Brake System": "Brake System",
+		"Interior Accessories": "Interior Accessories"
+	}
```
This is hard coded, can we make this dynamic too?

Altrix-One replied:
I have updated the migrate.py to remove the hardcoded elements, and also added it to the demo data script that can be installed from the Lifestyle Settings:

<img width="1470" height="277" alt="image" src="https://github.com/user-attachments/assets/c1fa0c99-8421-4635-adae-e127c92a8ee9" />


The rest is already dynamic and can be controlled from the Ecommerce Catergory doctype:
<img width="1470" height="318" alt="image" src="https://github.com/user-attachments/assets/e4d9dca2-2198-4e00-bd77-921b5018621e" />


Footers and Sections is also Dynamic now and can be controlled from the Footer Section Config: 
<img width="1470" height="339" alt="image" src="https://github.com/user-attachments/assets/041b255c-5b9d-400a-a2c0-4bde8bd825d6" />
<img width="1470" height="417" alt="image" src="https://github.com/user-attachments/assets/de2a5d44-0e30-4729-8425-e909da7b0e1a" />

Altrix-One replied:
Here is a demo site: https://shopping.cohenix.cloud/

Altrix-One committed:
Remove Hard Coded Elements

Altrix-One committed:
Colors now fully dynamic

Altrix-One committed:
feat(theme): Colors now fully dynamic

Altrix-One commented:
@Rl0007 Sounds good ! I have made some changes and have not updated the site for the final elements which were not dynamic. Will update that and let you know. I did not make ant changes to the default workflow or files related to that.

Altrix-One committed:
Update Final Colors

Altrix-One committed:
feat(theme): Update Final Colors

Altrix-One commented:
> Hi @Altrix-One this looks awesome 🌟🌟 We were thinking of doing something similar.. 😅
> 
> I am going through your demo site, can you please setup your email account and also share some test credentials. So I can checkout the complete flow till payment..
> 
> I also noticed some places where the colours are not dynamic. Can you also take a look into that..
> 
> <img alt="Screenshot 2025-10-07 at 4 20 23 PM" width="1470" height="827" src="https://private-user-images.githubusercontent.com/63764528/498260778-098abca1-3ec3-451a-859f-2a8a40dab557.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk4NDE4NzksIm5iZiI6MTc1OTg0MTU3OSwicGF0aCI6Ii82Mzc2NDUyOC80OTgyNjA3NzgtMDk4YWJjYTEtM2VjMy00NTFhLTg1OWYtMmE4YTQwZGFiNTU3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA3VDEyNTI1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTU3ZjUxYTA5MjMwMWVlMzUxODY4MzQ3Y2Q5YjJjYTAxNTU3OThiYzk3YjE3ZDBhM2IxZWI5ZjMzYjA4NGUwMGYmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.WHrPyIGl-eREKWMX8Yg0JzC3T41u9JNvRRFpA0FGz-Y"> <img alt="Screenshot 2025-10-07 at 4 20 31 PM" width="1470" height="827" src="https://private-user-images.githubusercontent.com/63764528/498260806-d93ef3be-4615-473f-9193-9ee435c09363.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk4NDE4NzksIm5iZiI6MTc1OTg0MTU3OSwicGF0aCI6Ii82Mzc2NDUyOC80OTgyNjA4MDYtZDkzZWYzYmUtNDYxNS00NzNmLTkxOTMtOWVlNDM1YzA5MzYzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA3VDEyNTI1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTA2MTFmNGJmZGE4YTBiNTI3NWFlMzQzMmIxOTc4ZmQ5NTgwOTg0N2UxZmQyNGM3ZTU1OWJlNDBjZmZjNzE5MzkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.izY19B0ZMQc6V3reWS4vwj-Kh1JOgWLvupYyrFPlySc"> <img alt="Screenshot 2025-10-07 at 4 20 40 PM" width="1470" height="827" src="https://private-user-images.githubusercontent.com/63764528/498260812-e5101329-d3df-40bc-9c8c-5552d7bfa3df.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NTk4NDE4NzksIm5iZiI6MTc1OTg0MTU3OSwicGF0aCI6Ii82Mzc2NDUyOC80OTgyNjA4MTItZTUxMDEzMjktZDNkZi00MGJjLTljOGMtNTU1MmQ3YmZhM2RmLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNTEwMDclMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjUxMDA3VDEyNTI1OVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY0ZWViZGQ3NGUzODlhM2Y5NGY3NDUzMWUwNDBkMWIxMGIyZDI0YmFhNDQ2YmNmZTA2MjExNTE0MzFlYTgzOTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.TWosuvZ18n8trK1BmOEoP_lZT5cIBV1ziGWWwi61Fko">
> Really looking forward to collaborate with you...

I have now updated the last few color elements:

<img width="1470" height="603" alt="image" src="https://github.com/user-attachments/assets/eab585ce-c097-4f13-8eec-f0bda6b43445" />
<img width="1470" height="665" alt="image" src="https://github.com/user-attachments/assets/272eec5d-30ba-40be-85fb-a186aa750ca4" />

Altrix-One committed:
Additional Colors added to be dynamic

Altrix-One committed:
feat(theme): Additional Colors added to be dynamic

Altrix-One committed:
Update hardcoded logo in search bar modal to use dynamic logo from Lifestyle Settings

Altrix-One committed:
fix(list): handle brands with no value to prevent errors

NagariaHussain commented:
Let's get this merged as soon as the tests pass @rl0007

Rl0007 committed:
Merge branch 'develop' into develop_cohenix

Altrix-One committed:
Merge branch 'BuildWithHussain:develop_cohenix' into develop_cohenix

Altrix-One committed:
fix(demo): add manual commit comments for demo data installation and publishing

Altrix-One committed:
Refactor: Fixed Pre-commit issues.

- Updated import statements in lifestyle_settings.py for better formatting.
- Cleaned up whitespace and formatting in various methods of lifestyle_settings.py.
- Enhanced error logging in migrate.py for better clarity on exceptions.
- Streamlined email template creation logic in migrate.py.
- Improved item publishing logic in publish_demo_items.py with consistent formatting.
- Removed unnecessary whitespace and improved readability in product listing logic in list.py.
